### PR TITLE
Implement DA client interface & Fix CI failure

### PR DESCRIPTION
**Pull request by steelgeek091**

### CODE REVIEW

File: rooch-da/lib.rs
Comment (jolestar) [inline rooch-da/lib.rs:1]: `rooch-da` is a crate. Please config the `Cargo.toml` and enable the build.

Comment (steelgeek091) [reply rooch-da/lib.rs:1]: > `rooch-da` is a crate. Please config the `Cargo.toml` and enable the build.

Got it, I am currently fixing this problem.

**Review by jolestar — CHANGES_REQUESTED** [review]

**Review by pause125 — APPROVED** [review]

File: rooch-da/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+pub struct DAGetResponse {
+    pub data: Vec<u8>,
+}
+
+pub struct DAStatusResponse {
+    pub status_code: u64,
+    pub status_data: Vec<u8>,
+}
+
+pub trait DAClientMethods {
```
Comment (templexxx) [inline rooch-da/src/lib.rs:41]: We've discussed about APIs here and in meetings:

https://github.com/rooch-network/rooch/issues/85

Pls trying to update codes according to results we've achieved

**Review by templexxx — CHANGES_REQUESTED** [review]

File: rooch-da/src/lib.rs (hunk 1 above)
Comment (steelgeek091) [reply rooch-da/src/lib.rs:34]: The problem has been fixed based on the previous records.

File: rooch-da/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+// Request to store data to the DA server
+pub struct DAStoreRequest {
+    pub kv: Vec<u8>,
+    pub checksum: u64,
+    pub signature: Vec<u8>,
+}
+
+// Data storage response returned by the DA server
+pub struct DAStoreResponse {
```
Comment (templexxx) [inline rooch-da/src/lib.rs:12]: should be put response

File: rooch-da/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,37 @@
+#[cfg(test)]
+mod test;
+
+// Request to store data to the DA server
+pub struct DAStoreRequest {
+    pub kv: Vec<u8>,
+    pub checksum: u64,
```
Comment (templexxx) [inline rooch-da/src/lib.rs:7]: Could comment here to tell others is application layer data integrity checking

**Review by templexxx — COMMENTED** [review]
LGTM in general 

Just a few issues, looking forward to ur fix

**Review by templexxx — APPROVED** [review]

**Review by jolestar — APPROVED** [review]
Need to rebase

**Pull request merged**